src/deadline/client/ui/dialogs/submit_job_progress_dialog.py

```diff
@@ -361,3 +357,123 @@ def _build_ui(self):
+        top_layout.addStretch()
+        layout.addLayout(top_layout)
+
+        # Scrollable Text Area
+        self.text_edit = QTextEdit()
+        self.text_edit.setPlainText(message)
+        self.text_edit.setReadOnly(True)
+        layout.addWidget(self.text_edit)
+
+        # Set minimum width and calculate optimal height
+        self.text_edit.setMinimumWidth(500)
+        self._calculate_optimal_size()
+
+        self.buttons = QDialogButtonBox(QDialogButtonBox.Ok | QDialogButtonBox.Cancel)
+        self.buttons.accepted.connect(self.accept)
+        self.buttons.rejected.connect(self.reject)
+
+        if default_response:
+            # If the default response is to continue, add the "Do not ask again" button
+            dont_ask_button = QPushButton("Do not ask again", self)
+            dont_ask_button.clicked.connect(lambda: set_setting("settings.auto_accept", "true"))
+            dont_ask_button.clicked.connect(self.accept)
+            self.buttons.addButton(dont_ask_button, QDialogButtonBox.ActionRole)
+
+        layout.addWidget(self.buttons)
+
+    def _calculate_optimal_size(self):
+        """Calculate optimal height based on text content and font metrics."""
+        # Get font metrics for the text edit
+        font_metrics = QFontMetrics(self.text_edit.font())
+
+        # Calculate text width accounting for margins and scrollbar
+        text_width = self.text_edit.minimumWidth() - 40  # Account for margins and scrollbar
+
+        # Calculate the height needed for the text
+        text_rect = font_metrics.boundingRect(0, 0, text_width, 0, Qt.TextWordWrap, self.message)
+        text_height = text_rect.height()
+
+        # Add some padding for better appearance
+        padding = 20
+        optimal_height = text_height + padding
+
+        # Set reasonable bounds
+        min_height = 100
+        max_height = 500
+
+        # Clamp the height to reasonable bounds
+        final_height = max(min_height, min(optimal_height, max_height))
+
+        self.text_edit.setMinimumHeight(final_height)
+        self.text_edit.setMaximumHeight(max_height)
+
+        # If content is short enough, disable scrollbar by setting max height to content height
+        if optimal_height <= max_height:
+            self.text_edit.setMaximumHeight(optimal_height)
+
+    def showEvent(self, event):
+        """Override showEvent to set the default button after the dialog is shown."""
+        super().showEvent(event)
+
+        # Set the default button after the dialog is fully shown
+        if self.buttons is not None:  # This type check is to make linting pass
+            if self.default_response:
+                ok_button = self.buttons.button(QDialogButtonBox.Ok)
+                if ok_button:
+                    ok_button.setDefault(True)
+                    ok_button.setAutoDefault(True)
+                    ok_button.setFocus()
+            else:
+                cancel_button = self.buttons.button(QDialogButtonBox.Cancel)
```

Comment:
I found when I select the cancel button, it submits the job instead of canceling.